### ISSUE TITLE
GooLeNet feature loading? 

### ISSUE THREAD

**Issue by astrude** [Russell91/TensorBox#120]
We are trying to parallelize the Reinspect model on multi-GPU machine, what we have observed is the GoogLenet features are loaded for every run/iteration? Is our understanding correct? 
How can we access the weights after every run so that this can be fed as input to data on multiple GPUs and gradients that are derived from each of the GPU will be synchronized via average which are fed into further iterations similar to [CIFAR-10 Multi-GPU ](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/models/image/cifar10/cifar10_multi_gpu_train.py) . Thanks in advance

**Comment by Russell91** [comment]
I don't think the GoogLeNet features are loaded every iteration. Can you point out the line where you think this is happening and why you think it occurs on every iteration?

**Comment by astrude** [comment]
For every iteration up until max iterations, Loss['train'] and train_op, this is where we believe the forward and backward function is called to perform those specific operations. In which case, for every new set of gradients generated for that iteration, the function will start from the beginning and compute gradients during which the weights are loaded again. This is just our understanding, it could be wrong. Could you please point out where the weights are taken as input for every new run and where the gradients output? This will be super helpful for us.

**Comment by Russell91** [comment]
The weights are loaded from line 470 of train.py:

            init_fn = slim.assign_from_checkpoint_fn(

For the rest, I suggest you go through the tensorflow tutorials and play around a little more. TensorBox is for the most part using the standard model suggested by the main tutorials.

**Comment by astrude** [comment]
Thanks!
